tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalCalendarData.java

```diff
@@ -0,0 +1,269 @@
+        private final String calendarType; // for comparison
+        private final GregorianCalendar startCalendar;
+        private final GregorianCalendar endCalendar;
+
+        EraData(final XPathValue xpath) {
+            calendarType = xpath.getAttributeValue(2, LDMLConstants.TYPE);
+            type = Integer.parseInt(xpath.getAttributeValue(INDEX, LDMLConstants.TYPE));
+
+            start = xpath.getAttributeValue(INDEX, LDMLConstants.START);
+            startCalendar = forDateString(start);
+            end = xpath.getAttributeValue(INDEX, LDMLConstants.END);
+            endCalendar = forDateString(end);
+            code = xpath.getAttributeValue(INDEX, LDMLConstants.CODE);
+        }
+
+        public int getType() {
+            return type;
+        }
+
+        public String getStart() {
+            return start;
+        }
+
+        public String getEnd() {
+            return end;
+        }
+
+        public GregorianCalendar getStartCalendar() {
+            return startCalendar;
+        }
+
+        public GregorianCalendar getEndCalendar() {
+            return endCalendar;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public String[] getAliases() {
+            final String s = xpath.getAttributeValue(INDEX, LDMLConstants.ALIASES);
+            if (s == null) return "".split(" "); // empty array
+            return s.split(" ");
+        }
+
+        @Override
+        public String toString() {
+            return String.format(
+                    "ERA %d, [%s-%s] code=%s, aliases=%s",
+                    getType(), getStart(), getEnd(), getCode(), String.join(",", getAliases()));
+        }
+
+        private static final GregorianCalendar forDateString(String ymd) {
+            if (ymd == null) return null;
+            int multiplier = 1;
+            if (ymd.startsWith("-")) {
+                multiplier = -1;
+                ymd = ymd.substring(1);
+            }
+            final String[] parts = ymd.split("-");
+            try {
+                return new GregorianCalendar(
+                        multiplier * Integer.parseInt(parts[0]),
+                        Integer.parseInt(parts[1]) - 1,
+                        Integer.parseInt(parts[2]));
+            } catch (NumberFormatException nfe) {
+                throw new IllegalArgumentException("While parsing date string " + ymd, nfe);
+            }
+        }
+
+        private GregorianCalendar getLatest() {
+            GregorianCalendar l = getEndCalendar();
+            if (l == null) l = getStartCalendar();
+            return l;
+        }
+
+        /** only works within the same cal system */
+        @Override
+        public int compareTo(EraData o) {
+            return ComparisonChain.start()
+                    .compare(calendarType, o.calendarType)
+                    .compare(getLatest(), o.getLatest())
+                    .compare(getType(), o.getType())
```

Comment:
```suggestion
                    .compare(getEnd(), o.getEnd())
```
Because of the above feature, you will never need the type for comparison; just the calendarType and end.